bot/cogs/temporary.py

```diff
@@ -0,0 +1,144 @@
+
+
+@commands.guild_only()
+class Temporary(commands.GroupCog, group_name='warnet-temp'):
+    def __init__(self, bot: WarnetBot) -> None:
+        self.bot = bot
+        self.db_pool = bot.get_db_pool()
+
+    @commands.Cog.listener()
+    async def on_connect(self) -> None:
+        self._check_temprole.start()
+
+    @app_commands.command(name='add', description='Adds a temprole to user')
+    @app_commands.describe(
+        user='User to add temprole to',
+        duration='Example: 1d2h3m4s',
+        role='Role to add',
+    )
+    async def add_temporary_role(
+        self,
+        interaction: Interaction,
+        user: discord.Member,
+        duration: str,
+        role: discord.Role,
+    ) -> None:
+        await interaction.response.defer()
+        if role.id == GiveawayConfig.BLACKLIST_ROLE_ID:
+            return await interaction.followup.send(
+                'Cannot add blacklist giveaway role!!', ephemeral=True
+            )
+
+        if not interaction.user.guild_permissions.manage_roles:
+            return await interaction.followup.send(
+                'You do not have permission to manage roles', ephemeral=True
+            )
+
+        try:
+            duration_seconds = parse_time_string(duration)
+            if duration_seconds < 60:
+                return await interaction.followup.send(
+                    'Duration must be at least 1 minute', ephemeral=True
+                )
+            if duration_seconds > 60 * 60 * 24 * 365:
+                return await interaction.followup.send(
+                    'Duration must be at most 1 year', ephemeral=True
+                )
+            await user.add_roles(role)
+        except ValueError:
+            return await interaction.followup.send('Invalid duration format', ephemeral=True)
+        except discord.HTTPException:
+            return await interaction.followup.send(f'Something went wrong', ephemeral=True)
+
+        logger.info(f'Added role {role} to user {user.id} for {duration_seconds} seconds')
+        total_duration = datetime.now(timezone.utc) + timedelta(seconds=duration_seconds)
+
+        async with self.db_pool.acquire() as conn:
+            in_table = await conn.fetchval(
+                'SELECT 1 FROM temp_role WHERE user_id = $1 AND role_id = $2',
+                user.id,
+                role.id,
+            )
+
+        async with self.db_pool.acquire() as conn:
+            if in_table:
+                await conn.execute(
+                    'UPDATE temp_role SET end_time = $1 WHERE user_id = $2 AND role_id = $3',
+                    total_duration,
+                    user.id,
+                    role.id,
+                )
+            else:
+                await conn.execute(
+                    'INSERT INTO temp_role (user_id, role_id, end_time) VALUES ($1, $2, $3)',
+                    user.id,
+                    role.id,
+                    total_duration,
+                )
```

Comment:
Why do we need to acquire db connection 2 times when we can call it for once?

Reply:
hmm i forgot about this, i will use ON CONFLICT for this